core/templates/pages/exploration-player-page/services/answer-classification.service.ts

```diff
@@ -16,159 +16,173 @@
-  function(
-      AlertsService, AnswerClassificationResultObjectFactory,
-      PredictionAlgorithmRegistryService, StateClassifierMappingService,
-      DEFAULT_OUTCOME_CLASSIFICATION, ENABLE_ML_CLASSIFIERS,
-      EXPLICIT_CLASSIFICATION,
-      INTERACTION_SPECS, STATISTICAL_CLASSIFICATION,
-      TRAINING_DATA_CLASSIFICATION) {
-    /**
-     * Finds the first answer group with a rule that returns true.
-     *
-     * @param {*} answer - The answer that the user has submitted.
-     * @param {array} answerGroups - The answer groups of the interaction. Each
-     *     answer group contains rule_specs, which is a list of rules.
-     * @param {object} defaultOutcome - The default outcome of the interaction.
-     * @param {function} interactionRulesService The service which contains the
-     *     explicit rules of that interaction.
-     *
-     * @return {object} An AnswerClassificationResult domain object.
-     */
-    var classifyAnswer = function(
-        answer, answerGroups, defaultOutcome, interactionRulesService) {
-      // Find the first group that contains a rule which returns true
-      // TODO(bhenning): Implement training data classification.
-      for (var i = 0; i < answerGroups.length; i++) {
-        for (var j = 0; j < answerGroups[i].rules.length; j++) {
-          var rule = answerGroups[i].rules[j];
-          if (interactionRulesService[rule.type](answer, rule.inputs)) {
-            return AnswerClassificationResultObjectFactory.createNew(
-              answerGroups[i].outcome, i, j, EXPLICIT_CLASSIFICATION);
-          }
+@Injectable({providedIn: 'root'})
+export class AnswerClassificationService {
+  constructor(
+      private alertsService: AlertsService,
+      private answerClassificationResultObjectFactory:
+        AnswerClassificationResultObjectFactory,
+      private appService: AppService,
+      private interactionSpecsService: InteractionSpecsService,
+      private predictionAlgorithmRegistryService:
+        PredictionAlgorithmRegistryService,
+      private stateClassifierMappingService: StateClassifierMappingService) {}
+
+  /**
+   * Finds the first answer group with a rule that returns true.
+   *
+   * @param answer - The answer that the user has submitted.
+   * @param answerGroups - The answer groups of the interaction. Each answer
+   *     group contains rule_specs, which is a list of rules.
+   * @param defaultOutcome - The default outcome of the interaction.
+   * @param interactionRulesService The service which contains the explicit
+   *     rules of that interaction.
+   *
+   * @return AnswerClassificationResult domain object.
+   */
+  private classifyAnswer(
+      answer, answerGroups: AnswerGroup[], defaultOutcome: Outcome,
+      interactionRulesService): AnswerClassificationResult {
+    // Find the first group that contains a rule which returns true
+    // TODO(bhenning): Implement training data classification.
+    for (var i = 0; i < answerGroups.length; i++) {
```

Comment:
Nit: Considering that we're using ES6 standards in this file, can you get rid of all the vars here?

Reply:
Turns out this is necessary since `i` is used in the loop, but I've still added a few other ES6-ish things.

Reply:
Cool, thanks @brianrodri!